From bc5d5e0eec0e7968c854b0a630eb5cc54cb822fa Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 25 Feb 2025 19:11:58 +0100
Subject: [PATCH 1608/1695] scripts: sign_rproc_fw: change the format of the
 encryption key

Store the encryption key in binary format instead of text format to be
more generic with keys generator such as openssl and STM32_KeyGen_CLI.

The value of the key is no more 0 but aligned with the aes_test_key
defined in the remoteproc_pta.c.

Fixes: 41190c54a0f5 ("scripts: sign_rproc_fw: Add support for AES CBC encryption")

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c5efdee984b3cf1ade32a03a2768bad2f62485bf scripts: sign_rproc_fw: change the format of the encryption key]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I31e0b8daf3bf422bfc1340cce7d230fd18e4a01c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/439981
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 keys/rproc_enc_test_key.bin |  1 +
 keys/rproc_enc_test_key.txt |  1 -
 scripts/sign_rproc_fw.py    | 10 ++++------
 3 files changed, 5 insertions(+), 7 deletions(-)
 create mode 100644 keys/rproc_enc_test_key.bin
 delete mode 100644 keys/rproc_enc_test_key.txt

diff --git a/keys/rproc_enc_test_key.bin b/keys/rproc_enc_test_key.bin
new file mode 100644
index 000000000..76def78bb
--- /dev/null
+++ b/keys/rproc_enc_test_key.bin
@@ -0,0 +1 @@
+*+±uÿè&mž1Å~“µàU“I¦šLOö6#3>û=
\ No newline at end of file
diff --git a/keys/rproc_enc_test_key.txt b/keys/rproc_enc_test_key.txt
deleted file mode 100644
index cd09bbf16..000000000
--- a/keys/rproc_enc_test_key.txt
+++ /dev/null
@@ -1 +0,0 @@
-0000000000000000000000000000000000000000000000000000000000000000
diff --git a/scripts/sign_rproc_fw.py b/scripts/sign_rproc_fw.py
index 75a596bc2..30f91bc7a 100755
--- a/scripts/sign_rproc_fw.py
+++ b/scripts/sign_rproc_fw.py
@@ -428,7 +428,7 @@ def get_args(logger):
                              'multiple times to add multiple TLVs.',
                         default=[], dest='plat_tlv')
     parser.add_argument('--enc_key', required=False,
-                        help='AES-256 encryption key (32 bytes)',
+                        help='AES-256 encryption key (32 bytes) in binary format',
                         dest='enc_key')
 
     parsed = parser.parse_args()
@@ -506,15 +506,13 @@ def main():
     if args.enc_key:
         # Read the AES key from the file
         try:
-            with open(args.enc_key, 'r') as key_file:
-                key_hex = key_file.read().strip()
-                enc_key = bytes.fromhex(key_hex)
-                utils._check_byteslike("key", enc_key)
+            with open(args.enc_key, 'rb') as key_file:
+                enc_key = key_file.read()
         except Exception as e:
             print(f"Error reading AES key file: {e}")
             sys.exit(1)
         if len(enc_key) not in (16, 24, 32):
-            print("Error: AES key must be 128, 192, or 256 bits")
+            print("Error: AES key size must be 128, 192, or 256 bits")
             sys.exit(1)
         enc_type = ENUM_ENCRYPT_TYPE['AES_CBC']
         tlv.add('ENCTYPE', enc_type.to_bytes(1, 'little'))
-- 
2.47.2

